src/molecules/Chip.tsx

```diff
@@ -1,34 +1,52 @@
 import React from 'react';
-import { StyleSheet, View, ViewStyle, Text } from 'react-native';
+import { FlexStyle, StyleProp, StyleSheet, View, ViewStyle, Text } from 'react-native';
 import { registeredTextStyle } from 'src/utils/textStyles';
 import { colors } from 'src/utils/colors';
 
 const styles = StyleSheet.create({
   Wrapper: {
-    alignSelf: 'flex-start', // the only way to make sure that the element won't grow
     paddingHorizontal: 4,
     borderRadius: 7,
     height: 14,
     backgroundColor: colors.tertiary, // default bgState maps to this
   } as ViewStyle,
-  Text: registeredTextStyle('micro', colors.ulisse, 'chip-text'),
 });
 
 export type ChipProps = {
   label: string;
-  bgState: 'default' | 'success' | 'error';
+  bgState: 'default' | 'success' | 'error' | 'ulisse' | 'brand';
+  alignSelf?: FlexStyle['alignSelf'];
 };
 
-export const ChipUnmemoized = (props: ChipProps) => (
-  <View
-    style={
-      props.bgState === 'default'
-        ? styles.Wrapper
-        : [styles.Wrapper, { backgroundColor: colors[props.bgState] }]
-    }
-  >
-    <Text style={styles.Text}>{props.label.toUpperCase()}</Text>
+const getWrapperStyle = (
+  bgState: ChipProps['bgState'],
+  alignSelf: ChipProps['alignSelf']
+): StyleProp<ViewStyle> => {
+  switch (bgState) {
+    case 'default':
+      return [styles.Wrapper, { alignSelf }];
+    default:
+      return [styles.Wrapper, { alignSelf, backgroundColor: colors[bgState] }];
+  }
+};
+
+const getTextStyle = (bgState: ChipProps['bgState']) => {
```

Comment:
if `bgState` is refactored to `bgColor`, all the relevant variables should be updated, too